operator/v1/types_console.go

```diff
@@ -375,6 +380,28 @@ const (
 	BrandROSA Brand = "ROSA"
 )
 
+// IngressCustomization allows cluster admin to configure alternative ingress for the console.
+type IngressCustomization struct {
+	// consoleURL is a URL to be used as the base console address.
+	// This field is intended for clusters without ingress capability,
+	// where access to routes is not possible.
```

Comment:
You need to explain the behaviour when these are set on clusters that have ingress, I believe you said the Ingress overrides this?

Is there any use case where having this set would be preferable over the Route's that exist? Should we instead allow these to be overrides so users could add secondary/preferred DNS not via the Routes?

Reply:
> I believe you said the Ingress overrides this?

No, it won't have any effect on a cluster with ingress. There was no use case for this so far.

> Is there any use case where having this set would be preferable over the Route's that exist?

No, not yet.

> Should we instead allow these to be overrides so users could add secondary/preferred DNS not via the Routes?

May be, one day when such a request would come.

Reply:
@spadgett : so you would prefer the new fields to be respected in any context, not only when the ingress is disabled?

Reply:
Rephrased from
> This field is intended for clusters without ingress capability

to:
> This field can be useful in clusters without ingress capability

To open the possibility for broader interpretation.

Reply:
Posted in slack, copying here. I think we do want them to take precedence, and will need to document that within the godoc, something along the lines of
```
This will take precedence over any existing routes configured into the cluster.
Please make sure that appropriate ingress is configured at this URL before configuring the value.
Failure to do so may result in the degradation of several operators.
```

Reply:
Rephrased again to reflect [Trevor's point](https://github.com/openshift/api/pull/1887#discussion_r1609449647):
```
    // consoleURL is a URL to be used as the base console address.
    // If not specified, the console route hostname will be used.
    // This field is required for clusters without ingress capability,
    // where access to routes is not possible.
    // Make sure that appropriate ingress is set up at this URL.
    // The console operator will monitor the URL and may go degraded
    // if it's unreachable for an extended period.                                                                                    
    // Must use the HTTPS scheme.
```